src/aind_behavior_experiment_launcher/behavior_launcher/slims_picker.py

```diff
@@ -0,0 +1,401 @@
+
+        return slims_client
+
+    def test_client_connection(self) -> bool:
+        """
+        Test client connection by querying mouse id. Ignore exception if mouse id does not exist
+
+        Returns:
+            Boolean if connection works
+
+        Raises:
+            Exception: If invalid credentials or error reading from slims
+        """
+
+        try:
+            self.slims_client.fetch_model(SlimsMouseContent, barcode="00000000")
+            logger.info("Successfully connected to Slims")
+            return True
+
+        except exceptions.SlimsAPIException as e:
+            logger.warning(f"Exception trying to read from Slims: {e}.\n")
+            return False
+
+    @property
+    def slims_mouse(self) -> SlimsMouseContent:
+        """
+        Returns slims mouse model being used to load session
+
+        Returns:
+            SlimsMouseContent: slims mouse model object
+        """
+
+        return self._slims_mouse
+
+    @property
+    def slims_session(self) -> SlimsBehaviorSession:
+        """
+        Returns slims session model being used to load task logic
+
+        Returns:
+           SlimsBehaviorSession: slims session model object
+        """
+
+        return self._slims_session
+
+    @property
+    def slims_rig(self) -> SlimsInstrument:
+        """
+        Returns slims instrument model being used to load rig configuration
+
+        Returns:
+           SlimsInstrument: slims instrument model object
+        """
+
+        return self._slims_rig
+
+    def add_waterlog(
+        self,
+        weight_g: float,
+        water_earned_ml: float,
+        water_supplement_delivered_ml: float,
+        water_supplement_recommended_ml: Optional[float] = None,
+    ) -> None:
+        """
+        Add waterlog event to Slims
+
+        Args:
+            weight_g (float): animal weight in grams
+            water_earned_ml (float): water earned during session in mL
+            water_supplement_delivered_ml (float): supplemental water given in session mL
+            water_supplement_recommended_ml (Optional[float]): optional recommended water amount
+
+        """
+
+        if self.launcher.session_schema is not None:
+            # create model
+            model = SlimsWaterlogResult(
+                mouse_pk=self._slims_mouse.pk,
+                date=self.launcher.session_schema.date,
+                weight_g=weight_g,
+                operator=", ".join(self.launcher.session_schema.experimenter),
+                water_earned_ml=water_earned_ml,
+                water_supplement_delivered_ml=water_supplement_delivered_ml,
+                water_supplement_recommended_ml=water_supplement_recommended_ml,
+                total_water_ml=water_earned_ml + water_supplement_delivered_ml,
+                comments=self.launcher.session_schema.notes,
+                workstation=self.launcher.rig_schema.name,
+                test_pk=self.slims_client.fetch_pk("Test", test_name="test_waterlog"),
```

Comment:
A RigModel (https://allenneuraldynamics.github.io/Aind.Behavior.Services/api.rig.html) does not have a name property. You can use the `computer_name` or `rig_name` whichever you think fits best with the model.